src/mdns/mdns.hpp

```diff
@@ -159,8 +159,9 @@ class Publisher : public MainloopProcessor
      */
     enum class State
     {
-        kIdle,  ///< Unable to publishing service.
-        kReady, ///< Ready for publishing service.
+        kIdle,      ///< Unable to publish service.
+        kReady,     ///< Ready to publish service.
+        kRestarted, ///< Recently restarted.
```

Comment:
Should we change the state to `kIdle` when disconnected from mDNS, and change back to `kReady` when mDNS is ready again?
My major concern is that `kRestarted` is similar to `kReady` state.